### DISCUSSION TITLE
R5s 安装 PassWall 失败

### DISCUSSION THREAD

**Discussion by JIUJIUs** [istoreos/istoreos#1194] [Q&A] [answered]
R5s 安装 PassWall 报错，请问是缺少的依赖包要去哪里安装？

root@iStoreOS:/tmp/pass# ls
brook_20230606-1_aarch64_cortex-a53.ipk                       shadowsocks-rust-sslocal_1.15.4-1_aarch64_cortex-a53.ipk      tuic-client_1.0.0-1_aarch64_cortex-a53.ipk
chinadns-ng_2023.06.01-1_aarch64_cortex-a53.ipk               shadowsocks-rust-ssserver_1.15.4-1_aarch64_cortex-a53.ipk     v2ray-core_5.7.0-1_aarch64_cortex-a53.ipk
dns2socks_2.1-2_aarch64_cortex-a53.ipk                        shadowsocksr-libev-ssr-local_2.5.6-9_aarch64_cortex-a53.ipk   v2ray-geoip_202302090046-1_all.ipk
dns2tcp_1.1.0-1_aarch64_cortex-a53.ipk                        shadowsocksr-libev-ssr-redir_2.5.6-9_aarch64_cortex-a53.ipk   v2ray-geosite_20230208103804-1_all.ipk
hysteria_1.3.5-1_aarch64_cortex-a53.ipk                       shadowsocksr-libev-ssr-server_2.5.6-9_aarch64_cortex-a53.ipk  v2ray-plugin_5.7.0-1_aarch64_cortex-a53.ipk
luci-app-passwall_4.68-5_all.ipk                              simple-obfs_0.0.5-1_aarch64_cortex-a53.ipk                    xray-core_1.8.3-1_aarch64_cortex-a53.ipk
luci-i18n-passwall-zh-cn_4.68-5_all.ipk                       tcping_0.3-1_aarch64_cortex-a53.ipk                           xray-plugin_1.8.3-1_aarch64_cortex-a53.ipk
microsocks_1.0.3-1_aarch64_cortex-a53.ipk                     trojan-go_0.10.6-1_aarch64_cortex-a53.ipk
naiveproxy_113.0.5672.62-1-1_aarch64_cortex-a53.ipk           trojan-plus_10.0.3-2_aarch64_cortex-a53.ipk
root@iStoreOS:/tmp/pass# opkg install *.ipk
Installing brook (20230606-1) to root...
Installing chinadns-ng (2023.06.01-1) to root...
Installing dns2socks (2.1-2) to root...
Installing dns2tcp (1.1.0-1) to root...
Installing hysteria (1.3.5-1) to root...
Unknown package 'luci-app-passwall'.
Installing luci-i18n-passwall-zh-cn (4.68-5) to root...
Installing microsocks (1.0.3-1) to root...
Installing naiveproxy (113.0.5672.62-1-1) to root...
Installing shadowsocks-rust-sslocal (1.15.4-1) to root...
Installing shadowsocks-rust-ssserver (1.15.4-1) to root...
Unknown package 'shadowsocksr-libev-ssr-local'.
Unknown package 'shadowsocksr-libev-ssr-redir'.
Unknown package 'shadowsocksr-libev-ssr-server'.
Unknown package 'simple-obfs'.
Installing tcping (0.3-1) to root...
Installing trojan-go (0.10.6-1) to root...
Unknown package 'trojan-plus'.
Installing tuic-client (1.0.0-1) to root...
Installing v2ray-core (5.7.0-1) to root...
Installing v2ray-geoip (202302090046-1) to root...
Installing v2ray-geosite (20230208103804-1) to root...
Installing v2ray-plugin (5.7.0-1) to root...
Installing xray-core (1.8.3-1) to root...
Installing xray-plugin (1.8.3-1) to root...
Configuring tuic-client.
Configuring hysteria.
Configuring chinadns-ng.
Configuring shadowsocks-rust-ssserver.
Configuring dns2socks.
Configuring trojan-go.
Configuring naiveproxy.
Configuring dns2tcp.
Configuring v2ray-plugin.
Configuring microsocks.
Configuring xray-plugin.
Configuring shadowsocks-rust-sslocal.
Configuring tcping.
Configuring v2ray-core.
Configuring brook.
Configuring v2ray-geosite.
Configuring v2ray-geoip.
Configuring xray-core.
Collected errors:
 * pkg_hash_check_unresolved: cannot find dependency coreutils-base64 for luci-app-passwall
 * pkg_hash_check_unresolved: cannot find dependency coreutils-nohup for luci-app-passwall
 * pkg_hash_check_unresolved: cannot find dependency resolveip for luci-app-passwall
 * pkg_hash_fetch_best_installation_candidate: Packages for luci-app-passwall found, but incompatible with the architectures configured
 * opkg_install_cmd: Cannot install package luci-app-passwall.
 * satisfy_dependencies_for: Cannot satisfy the following dependencies for luci-i18n-passwall-zh-cn:
 *      coreutils-base64
 *      coreutils-nohup
 *      resolveip
 * opkg_install_cmd: Cannot install package luci-i18n-passwall-zh-cn.
 * pkg_hash_check_unresolved: cannot find dependency libev for shadowsocksr-libev-ssr-local
 * pkg_hash_check_unresolved: cannot find dependency libsodium for shadowsocksr-libev-ssr-local
 * pkg_hash_check_unresolved: cannot find dependency libudns for shadowsocksr-libev-ssr-local
 * pkg_hash_fetch_best_installation_candidate: Packages for shadowsocksr-libev-ssr-local found, but incompatible with the architectures configured
 * opkg_install_cmd: Cannot install package shadowsocksr-libev-ssr-local.
 * pkg_hash_check_unresolved: cannot find dependency libev for shadowsocksr-libev-ssr-redir
 * pkg_hash_check_unresolved: cannot find dependency libsodium for shadowsocksr-libev-ssr-redir
 * pkg_hash_check_unresolved: cannot find dependency libudns for shadowsocksr-libev-ssr-redir
 * pkg_hash_fetch_best_installation_candidate: Packages for shadowsocksr-libev-ssr-redir found, but incompatible with the architectures configured
 * opkg_install_cmd: Cannot install package shadowsocksr-libev-ssr-redir.
 * pkg_hash_check_unresolved: cannot find dependency libev for shadowsocksr-libev-ssr-server
 * pkg_hash_check_unresolved: cannot find dependency libsodium for shadowsocksr-libev-ssr-server
 * pkg_hash_check_unresolved: cannot find dependency libudns for shadowsocksr-libev-ssr-server
 * pkg_hash_fetch_best_installation_candidate: Packages for shadowsocksr-libev-ssr-server found, but incompatible with the architectures configured
 * opkg_install_cmd: Cannot install package shadowsocksr-libev-ssr-server.
 * pkg_hash_check_unresolved: cannot find dependency libev for simple-obfs
 * pkg_hash_fetch_best_installation_candidate: Packages for simple-obfs found, but incompatible with the architectures configured
 * opkg_install_cmd: Cannot install package simple-obfs.
 * pkg_hash_check_unresolved: cannot find dependency boost for trojan-plus
 * pkg_hash_check_unresolved: cannot find dependency boost-system for trojan-plus
 * pkg_hash_check_unresolved: cannot find dependency boost-program_options for trojan-plus
 * pkg_hash_fetch_best_installation_candidate: Packages for trojan-plus found, but incompatible with the architectures configured
 * opkg_install_cmd: Cannot install package trojan-plus.

**Comment by JIUJIUs** [comment, marked as answer]
暂时解决了，需要到 https://downloads.openwrt.org/snapshots/packages/aarch64_cortex-a53/ 下载依赖
<img width="406" alt="image" src="https://github.com/istoreos/istoreos/assets/61548656/c44e448c-a2af-45ed-aebc-30f15c339987">

**Comment by jjm2473** [comment]
https://github.com/istoreos/istoreos/issues/1057
